app/models/podcast_import.rb

```diff
@@ -13,89 +13,160 @@ class PodcastImport < BaseModel
-    elsif episode_imports.any? { |e| e.status == 'failed' }
-      update_attributes!(status: 'failed')
+  def status
+    return super unless episode_imports.count > 0
+    return super if feed_episode_count > episode_imports.count
+
+    if complete?
+      COMPLETE
+    elsif finished? && some_failed?
+      FAILED
+    else
+      super
     end
   end
 
+  def finished?
+    return false unless episode_imports.count == feed_episode_count
+    episode_imports.all? do |e|
+      e.status == EpisodeImport::COMPLETE ||
+        e.status == EpisodeImport::FAILED
+    end
+  end
+
+  def complete?
+    return false unless episode_imports.count == feed_episode_count
+    episode_imports.all? { |e| e.status == EpisodeImport::COMPLETE }
+  end
+
+  def some_failed?
+    episode_imports.any? { |e| e.status == EpisodeImport::FAILED }
+  end
+
   def retry!
-    update_attributes(status: 'retrying')
+    update_attributes(status: RETRYING)
     import_later
   end
 
-  def import_later
-    PodcastImportJob.perform_later self
+  def import_later(import_series = true)
+    PodcastImportJob.perform_later(self, import_series)
   end
 
-  def import
-    update_attributes!(status: 'started')
+  def import_series!
+    update_attributes!(status: STARTED)
 
     # Request the RSS feed
     get_feed
-    update_attributes!(status: 'feed retrieved')
+    update_attributes!(status: FEED_RETRIEVED)
 
     # Create the series
     create_or_update_series!
-    update_attributes!(status: 'series created')
+    update_attributes!(status: SERIES_CREATED)
+  rescue StandardError => err
+    update_attributes(status: FAILED)
+    raise err
+  end
+
+  def import_episodes!
+    # Request the RSS feed again
+    get_feed
 
     # Update podcast attributes
     create_or_update_podcast!
-    update_attributes!(status: 'podcast created')
+    update_attributes!(status: PODCAST_CREATED)
 
     # Create the episodes
-    update_attributes!(status: 'importing')
+    update_attributes!(status: IMPORTING)
     create_or_update_episode_imports!
   rescue StandardError => err
-    update_attributes(status: 'failed')
+    update_attributes(status: FAILED)
     raise err
   end
 
-  def entry_audio_files(entry)
-    if config[:audio] && config[:audio][entry[:entry_id]]
-      { files: (config[:audio][entry[:entry_id]] || []) }
-    elsif enclosure = enclosure_url(entry)
-      { files: [enclosure] }
-    end
+  def import
+    import_series!
+    import_episodes!
   end
 
   def create_or_update_episode_imports!
-    feed.entries.map do |entry|
-      entry_hash = feed_entry_to_hash(entry)
-      audio_files = entry_audio_files(entry_hash)
-      get_or_create_template(audio_files, entry_hash['enclosure'].type)
-      episode_import = create_or_update_episode_import!(entry_hash, audio_files)
-      episode_import.import_later
+    feed_entries, entries_with_dupe_guids = parse_feed_entries_for_dupe_guids
+
+    update_attributes(feed_episode_count: feed_entries.length)
```

Comment:
Wait, so then `feed_episode_count` _won't_ be the number of `<item>`s in the RSS?  Seems like this is going to un-count any items with duplicate guids.

Reply:
Yep, that's a mistake! Thanks for catching.  When I renamed that variable, I shifted the semantics, so I need to adjust my arithmetic here and over in publish:
https://github.com/PRX/publish.prx.org/pull/538/files#diff-92016921774e9e471370aba107bc8249R62